sagemathinc/cocalc
# Fix sign up text insertion

# Description
Fix invalid email bug when a browser or extension inserts text which doesn't trigger onChange.
Also Typescript Passports and SignUp.

The **last commit** is the actual work to fix the sign up bug.

I hope following the Typescripting commits is fairly straightforward.

# Testing Steps
1. Inspect the form elements
1. Insert `value="sneaky@email.com"` into the email input element
1. See it appear on the form
1. Test the same insert method for the rest of the form
1. Create the account
1. It should be created

# Relevant Issues

### [Checklist](https://github.com/sagemathinc/cocalc/wiki/PR-Checklist):
- Screenshots not relevant
